include/envoy/http/codec.h

```diff
@@ -354,6 +354,19 @@ class Stream {
   virtual void setFlushTimeout(std::chrono::milliseconds timeout) PURE;
 };
 
+/**
+ * A class for sharing what HTTP/1 SETTINGS were received from the peer.
+ */
+class ReceivedSettings {
```

Comment:
It does not look like you are actually gaining anything by having this interface. In tests you can just construct `ReceivedSettingImpl`. Might be worthwhile to make it a concrete class without the need for the interface, since it only carries state and no behavior.